File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
       unordered_set<char> k;
        int low=0;
        int high=0;
        int max_count=0;
       while(high<s.length())
       {
        if(k.find(s[high])!=k.end()){
            max_count=max(max_count,high-low);
            k.erase(s[low]);
            low++;
            

        }
        else{
            k.insert(s[high]);
            high++;
           
        }
       }
       max_count=max(max_count,high-low);
       return max_count;

        
    }
};

File: Data Structures & Algorithms/search-2d-matrix/submission-1.py
class Solution:     
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        low=0
        high=(len(matrix)*len(matrix[0]))-1
        n=len(matrix[0])
        
        while(low<=high):
                mid=low+(high-low)//2

                if matrix[mid//n][mid%n]==target:
                    return True
                elif matrix[mid//n][mid%n]>target:
                    high=mid-1
                else:
                    low=mid+1
        return False
        

            
        